test/trafficGenerator/buildPreSpawnAircraft.spec.js

```diff
@@ -1,42 +1,200 @@
+    const totalDistanceMock = 60; // 10 minutes to airspace boundary (at 6mpm)
+    const offsetDistanceMock = 18; // 7 minute to airspace boundary (at 6mpm)
+    const expectedResult = airspaceCeilingMock + 7000; // 1000fpm descent rate for 1 minute
+    const result = _calculateIdealSpawnAltitudeAtOffset(altitudesAtOffsets, offsetDistanceMock, spawnSpeedMock, spawnAltitudeMock, totalDistanceMock, airspaceCeilingMock);
+
+    t.true(expectedResult === result);
+});
+
+ava('_calculateIdealSpawnAltitudeAtOffset() returns an interpolated altitude based on available descent distance to first altitude restriction', (t) => {
+    const altitudesAtOffsets = [
+        [18.958610430426404, 19000],
+        [41.52033243401482, 12000],
+        [60.46996764011041, 10000],
+        [70.68723901280046, 8000]
+    ];
+    const spawnAltitudeMock = 23000;
+    const airspaceCeilingMock = 11000;
+    const spawnSpeedMock = 360;
+    const totalDistanceMock = 85;
+    const offsetDistanceMock = 6.5;
+    const expectedResult = 21000;
+    const result = _calculateIdealSpawnAltitudeAtOffset(altitudesAtOffsets, offsetDistanceMock, spawnSpeedMock, spawnAltitudeMock, totalDistanceMock, airspaceCeilingMock);
+
+    t.true(expectedResult === result);
+});
+
+ava('_calculateIdealSpawnAltitudeAtOffset() returns an interpolated altitude based on a flat glidepath between restrictions when spawn point is beyond the first altitude restriction', (t) => {
+    const altitudesAtOffsets = [
+        [18.958610430426404, 19000],
+        [41.52033243401482, 12000],
+        [60.46996764011041, 10000],
+        [70.68723901280046, 8000]
+    ];
+    const spawnAltitudeMock = 23000;
+    const airspaceCeilingMock = 11000;
+    const spawnSpeedMock = 360;
+    const totalDistanceMock = 85;
+    const offsetDistanceMock = 25;
+    const expectedResult = 17000;
+    const result = _calculateIdealSpawnAltitudeAtOffset(altitudesAtOffsets, offsetDistanceMock, spawnSpeedMock, spawnAltitudeMock, totalDistanceMock, airspaceCeilingMock);
+
+    t.true(expectedResult === result);
+});
+
+ava('_calculateIdealSpawnAltitudeAtOffset() returns an appropriate altitude when a range of spawn altitudes is specified instead of a specific one', (t) => {
+    const altitudesAtOffsets = [
+        [18.958610430426404, 19000],
+        [41.52033243401482, 12000],
+        [60.46996764011041, 10000],
+        [70.68723901280046, 8000]
+    ];
+    const spawnAltitudeMock = [23000, 23456];
+    const airspaceCeilingMock = 11000;
+    const spawnSpeedMock = 250;
+    const totalDistanceMock = 85;
+    const offsetDistanceMock = 0;
+    const expectedResult = 23000;
+    const result = _calculateIdealSpawnAltitudeAtOffset(altitudesAtOffsets, offsetDistanceMock, spawnSpeedMock, spawnAltitudeMock, totalDistanceMock, airspaceCeilingMock);
+
+    t.true(expectedResult === result);
+});
+
+// ava.skip('buildPreSpawnAircraft() throws when passed invalid parameters', (t) => {
```

Comment:
if these were already skipped and now they're commented out, let's just remove them

Reply:
Sorry, they're future tests that are needed, they just weren't working, so I commented them out. I'll leave them there when we get a chance to get it up to full coverage, because they may have good stuff in them.